src/main/java/com/iexec/common/docker/client/DockerClientInstance.java

```
@@ -116,11 +120,15 @@ public boolean isVolumePresent(String volumeName) {
         }
     }
 
-    public boolean removeVolume(String volumeName) {
+    public synchronized boolean removeVolume(String volumeName) {
         if (StringUtils.isBlank(volumeName)) {
             return false;
         }
-        try (RemoveVolumeCmd removeVolumeCmd = this.client.removeVolumeCmd(volumeName)) {
+        if (!isVolumePresent(volumeName)) {
+            log.info("No docker volume to remove [name:{}]", volumeName);
+            return false;
```

Unsure if we want true or false

---

At first, I also thought it would be better to return `true` but the problem is whenever an error occurs in the method `isVolumePresent()` (docker exception for example) we will return `true` without removing the volume.

---

Ok great :+1:

---

Same thing for networks, containers, and images.

---

In the future, I think we should maybe let the exceptions propagate to the caller.

---

I agree